services/bots/__init__.py

```diff
@@ -91,43 +139,33 @@ def _get_repo_appropriate_bot(repo: Repository) -> Owner:
             extra=dict(repoid=repo.repoid, botid=repo.bot.ownerid),
         )
         return repo.bot
-    if repo.owner.bot is not None and repo.owner.bot.oauth_token is not None:
-        log.info(
-            "Repo Owner has specific bot",
-            extra=dict(
-                repoid=repo.repoid,
-                botid=repo.owner.bot.ownerid,
-                ownerid=repo.owner.ownerid,
-            ),
-        )
-        return repo.owner.bot
-    if repo.owner.oauth_token is not None:
-        log.info(
-            "Using repo owner as bot fallback",
-            extra=dict(repoid=repo.repoid, ownerid=repo.owner.ownerid),
-        )
-        return repo.owner
-    raise RepositoryWithoutValidBotError()
+    try:
+        return _get_owner_or_appropriate_bot(repo.owner)
+    except OwnerWithoutValidBotError:
+        raise RepositoryWithoutValidBotError()
 
 
-def get_owner_appropriate_bot_token(
+def _get_owner_appropriate_bot_token(
     owner, installation_info: GithubInstallationInfo | None = None
-) -> Token:
+) -> TokenWithOwner:
     if installation_info:
         result = get_github_app_token(Service(owner.service), installation_info)
-        return result[0]
+        return result
 
-    return encryptor.decrypt_token(_get_owner_or_appropriate_bot(owner).oauth_token)
+    token_owner = _get_owner_or_appropriate_bot(owner)
+    return encryptor.decrypt_token(token_owner.oauth_token), token_owner
 
 
-def _get_owner_or_appropriate_bot(owner: Owner) -> Owner:
+def _get_owner_or_appropriate_bot(owner: Owner, repoid: int | None = None) -> Owner:
```

Comment:
does anybody use this new extra argument? cmd-f doesn't show usages

Reply:
It's used for logging.

I changed `_get_repo_appropriate_bot` (you can check the original version in the deleted `bots.py` file), and they had a slightly different message + the repoid in the logs.
I thought it would be important for debugging to keep the `repoid` in the logs